sds_data_manager/stacks/sds_data_manager_stack.py

```diff
@@ -0,0 +1,215 @@
+
+        # This is the S3 bucket used by upload_api_lambda
+        data_bucket = s3.Bucket(
+            self, f"DataBucket-{sds_id}",
+            bucket_name=f"sds-data-{sds_id}",
+            versioned=True,
+            removal_policy=RemovalPolicy.DESTROY,
+            auto_delete_objects=True,
+            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
+        )
+
+        # Confirm that a config.json file exists in the expected
+        # location before S3 upload
+        if (
+            not pathlib.Path(__file__)
+            .parent.joinpath("..", "config", "config.json")
+            .resolve()
+            .exists()
+        ):
+            raise RuntimeError(
+                "sds_data_manager/config directory must contain config.json"
+            )
+
+        # S3 bucket where the configurations will be stored
+        config_bucket = s3.Bucket(
+            self,
+            f"ConfigBucket-{sds_id}",
+            bucket_name=f"sds-config-bucket-{sds_id}",
+            versioned=True,
+            removal_policy=RemovalPolicy.DESTROY,
+            auto_delete_objects=True,
+            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
+        )
+
+        # Upload all files in the config directory to the S3 config bucket.
+        # This directory should contain a config.json file that will
+        # be used for indexing files into the data bucket.
+        s3_deploy.BucketDeployment(
+            self,
+            f"DeployConfig-{sds_id}",
+            sources=[
+                s3_deploy.Source.asset(
+                    str(pathlib.Path(__file__).parent.joinpath("..", "config").resolve())
+                )
+            ],
+            destination_bucket=config_bucket,
+        )
+
+        s3_write_policy = iam.PolicyStatement(
+            effect=iam.Effect.ALLOW,
+            actions=["s3:PutObject"],
+            resources=[f"{data_bucket.bucket_arn}/*"],
+        )
+        s3_read_policy = iam.PolicyStatement(
+            effect=iam.Effect.ALLOW,
+            actions=["s3:GetObject"],
+            resources=[f"{data_bucket.bucket_arn}/*", f"{config_bucket.bucket_arn}/*"],
+        )
+        iam.PolicyStatement(
+            effect=iam.Effect.ALLOW,
+            actions=["cognito-idp:*"],
+            resources=["*"],
+        )
+
+        indexer_lambda = lambda_alpha_.PythonFunction(
+            self,
+            id="IndexerLambda",
+            function_name=f"file-indexer-{sds_id}",
+            entry=str(pathlib.Path(__file__).parent.joinpath("..", "lambda_code").resolve()),
+            index="SDSCode/indexer.py",
+            handler="lambda_handler",
+            runtime=lambda_.Runtime.PYTHON_3_9,
+            timeout=cdk.Duration.minutes(15),
+            memory_size=1000,
+            environment={
+                "OS_ADMIN_USERNAME": "master-user",
+                "OS_ADMIN_PASSWORD_LOCATION": opensearch.os_secret.secret_value.unsafe_unwrap(),
```

Comment:
OS_ADMIN_PASSWORD_LOCATION can be deleted since you're using the secrets manager client in the lambda now